### PR TITLE
Light Enemy (In Progress)

### PR DESCRIPTION
**Pull request by JoshCTFS**
Light Enemy Script (Not Yet Tested)
Temp Med Enemy Script

Temp LightEnemy Prefab made without animations

Renamed some of the sprites appropriately (Subject to change)

Also have made New Layer labeled "Player" for ease of visibility as well as other functions.

### CODE REVIEW

File: Assets/_Scripts/Enemy/LightEnemy.cs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            strafeTimer = 10f;
+        }
+    }
+
+    void OnTriggerEnter(Collider other)
+        {
+            if (other.CompareTag("PlayerProjectile"))
+            {
+                damageTaken++;
+                if (damageTaken >= currentHealth)
```
Comment (MV1090) [inline Assets/_Scripts/Enemy/LightEnemy.cs]: This logic is taken care of in the Enemy script, on collision you just need to call TakeDamage() and input the amount of damage taken.

File: Assets/_Scripts/Enemy/LightEnemy.cs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+                {
+                Destroy(obj: LightEnemy);
```
Comment (MV1090) [inline Assets/_Scripts/Enemy/LightEnemy.cs]: The enemy is destroyed from the  EnemyDeath(input score value) method which needs to be called as an animation event at the end of the death animation.

File: Assets/_Scripts/Enemy/MediumEnemy.cs [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+            moveSpeed *= -1; //will change direction of the unit when timer elapses
+            strafeTimer = 10f;
+        }
+    }
+
+    void OnTriggerEnter(Collider other)
+    {
+        if (other.CompareTag("PlayerProjectile"))
+        {
+            damageTaken++;
```
Comment (MV1090) [inline Assets/_Scripts/Enemy/MediumEnemy.cs]: Same as above

File: Assets/_Scripts/Enemy/MediumEnemy.cs [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+            if (damageTaken >= currentHealth)
+            {
+                Destroy(obj: MediumEnemy);
```
Comment (MV1090) [inline Assets/_Scripts/Enemy/MediumEnemy.cs]: Same as above

**Review by MV1090 — COMMENTED** [review]
Good work bud, you just need to add the Shoot() method, and the enemy killed method to the end of the death animation when its in. I like the strafing I completely forgot about this, as its the same for all the enemy's it might be better off on the Enemy class so we don't need to repeat it on all the scripts..... Ziyad what do you think of this idea? or is the strafing taken care of somewhere else?

**Comment by SolomanDavis** [comment]
@MV1090 @JoshCTFS re: strafing
This strafing code is going to be super useful to be used in combination with the enemy pathfinding hover state. (see https://github.com/SolomanDavis/VGP145-S24-capstone-retro/pull/35).

However few things to note:
When the enemy is in the `EnemyPathfinding.Hover` state, the enemy will simply move towards its assigned `HoverLocation`. This means that the enemy itself is not the entity doing the strafing. Rather it is the assigned `HoverLocation` and its container (the upcoming rank grid) that will be doing the strafing and where this code should be applied.

In terms of where the strafing code should specifically live - I believe it should live in an `EnemyRankGrid` class that I'll be implementing today with @Armaan059 since it's the grid itself that will strafe, while the enemies are sticking to the assigned slots within the grid.

**Comment by JoshCTFS** [comment]
> @MV1090 @JoshCTFS re: strafing This strafing code is going to be super useful to be used in combination with the enemy pathfinding hover state. (see #35).
> 
> However few things to note: When the enemy is in the `EnemyPathfinding.Hover` state, the enemy will simply move towards its assigned `HoverLocation`. This means that the enemy itself is not the entity doing the strafing. Rather it is the assigned `HoverLocation` and its container (the upcoming rank grid) that will be doing the strafing and where this code should be applied.
> 
> In terms of where the strafing code should specifically live - I believe it should live in an `EnemyRankGrid` class that I'll be implementing today with @Armaan059 since it's the grid itself that will strafe, while the enemies are sticking to the assigned slots within the grid.

Ah, my apologies. oversight on my part! Thanks for the input!

**Review by MV1090 — APPROVED** [review]

**Pull request merged**